File: easy/CheckAsBeforeBs.js
// Given a string s consisting of only the characters 'a' and 'b', return true if every 'a' appears before every 'b' in the string. Otherwise, return false.

var checkString = function(s) {
  let hasB = false;
  for (let i = 0; i < s.length; i++) {
    if (hasB) {
      if (s[i] === 'a') {
        return false;
      }
    } else {
      if (s[i] === 'b') {
        hasB = true;
      }
    }
  }
  return true
};


console.log(checkString('aaaba'))

File: easy/CheckAsBeforeBs.py
# Given a string s consisting of only the characters 'a' and 'b', return true if every 'a' appears before every 'b' in the string. Otherwise, return false.

def checkString(s) :
  hasB = False
  for letter in s :
    if letter == 'a' and hasB == True :
      return False
    if letter == 'b' :
      hasB = True
  return True

print(checkString('aaaabbbbab'))